keysign/avahiwormholediscover.py

```diff
@@ -0,0 +1,45 @@
+import logging
+
+from .wormholereceive import WormholeReceive
+from .avahidiscovery import AvahiKeysignDiscoveryWithMac
+from .util import is_code_complete, parse_barcode
+
+log = logging.getLogger(__name__)
+
+
+class AvahiWormholeDiscover:
+    def __init__(self, userdata, discovery=None, callback=None, app_id=None):
+        # Check if the given code is a barcode
+        parsed = parse_barcode(userdata).get("WORM", [None])[0]
+        if parsed:
+            # In this way if we have a barcode we directly only use the wormhole code.
+            # Maybe we should let the users choose the preferred method or try with one
+            # by default and fallback to the other if it fails
+            self.userdata = parsed
+        else:
+            self.userdata = userdata
+        self.callback = callback
+        self.app_id = app_id
+        if discovery:
+            self.discovery = discovery
+        else:
+            self.discovery = AvahiKeysignDiscoveryWithMac()
+        self.worm = None
+
+    def start(self):
+        # if the code may be a valid wormhole one we try to use wormhole.
+        # Otherwise we use avahi
```

Comment:
hrm. what if the sender offers the key through both Avahi and via wormhole, but the receiver cannot connect to the Internet but only to the local network?  Would we reduce functionality with this patch?
Can we try to locate the key locally first and then try the Internet? Or maybe start trying both at the same time and hope that one will succeed?
This probably requires to not discard the avahi data during `__init__`.